pytorch_lightning/utilities/cloud_io.py

```diff
@@ -12,15 +12,27 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
 import io
 from distutils.version import LooseVersion
 from pathlib import Path
 from typing import IO, Union
 
 import fsspec
+from fsspec.implementations.local import LocalFileSystem
+
 import torch
 
 
+class LightningLocalFileSystem(LocalFileSystem):
+    """Extension of ``fsspec.implementations.local.LocalFileSystem`` where ``LightningLocalFileSystem.isdir`` behaves
+    the same as ``os.isdir``.
+    """
+
+    def isdir(self, path: str) -> bool:
```

Comment:
Could you explain this, why we have to provide this implementation?
I followed the code of ffsspec LocalFileSystem

https://github.com/intake/filesystem_spec/blob/891fb609767f0ddad8b24520335654ff3c13302f/fsspec/spec.py#L625

And it calls info which itself checks for symlink.

Reply:
Sure, ffsspec `LocalFileSystem.info` checks for links, but the returns `type = 'link'` here: https://github.com/intake/filesystem_spec/blob/891fb609767f0ddad8b24520335654ff3c13302f/fsspec/implementations/local.py#L59

But `AbstractFileSystem.isdir` returns `true` only if `type = 'directory'`. So in this case `isdir` returns false even though it is a directory (just also a sym link).

Reply:
is this something that should be fixed in fsspec upstream?

Reply:
@ananthsub Yeah, would be better if this was fixed upstream. I have opened an issue here: https://github.com/intake/filesystem_spec/issues/591 - will remove this fixed once / if that gets adressed :smiley: